model/configuration.py

```diff
@@ -769,3 +774,567 @@ def excluded_audio_data_sources(cls, _db):
+        """
+        return self._key
+
+    @property
+    def label(self):
+        """Returns option's label
+
+        :return: Option's label
+        :rtype: string
+        """
+        return self._label
+
+    def to_settings(self):
+        """Returns a dictionary containing option metadata in the SETTINGS format
+
+        :return: Dictionary containing option metadata in the SETTINGS format
+        :rtype: Dict
+        """
+        return {
+            'key': self.key,
+            'label': self.label
+        }
+
+
+class ConfigurationMetadataOwner(object):
+    """Interface representing class containing ConfigurationMetadata properties"""
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def get_setting_value(self, setting_name):
+        """Returns a settings'value
+
+        :param setting_name: Name of the setting
+        :type setting_name: string
+
+        :return: Setting's value
+        :rtype: Any
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def set_setting_value(self, setting_name, setting_value):
+        """Sets setting's value
+
+        :param setting_name: Name of the setting
+        :type setting_name: string
+
+        :param setting_value: New value of the setting
+        :type setting_value: Any
+        """
+        raise NotImplementedError()
+
+
+class ConfigurationMetadata(object):
+    """Contains configuration metadata"""
+
+    def __init__(self, key, label, description, type, required=False, default=None, options=None, category=None):
+        """Initializes a new instance of ConfigurationMetadata class
+
+        :param key: Setting's key
+        :type key: string
+
+        :param label: Setting's label
+        :type label: string
+
+        :param description: Setting's description
+        :type description: string
+
+        :param type: Setting's type
+        :type type: ConfigurationAttributeType
+
+        :param required: Boolean value indicating whether the setting is required or not
+        :type required: bool
+
+        :param default: Setting's default value
+        :type default: Any
+
+        :param options: Setting's options (used in the case of select)
+        :type options: List[ConfigurationSettingAttributeOption]
+
+        :param category: Setting's category
+        :type category: string
+        """
+        self._key = key
+        self._label = label
+        self._description = description
+        self._type = type
+        self._required = required
+        self._default = default
+        self._options = options
+        self._category = category
+
+    def __get__(self, owner_instance, owner_type):
+        """Returns a value of the setting
+
+        :param owner_instance: Instance of the owner, class having instance of ConfigurationMetadata as an attribute
+        :type owner_instance: Optional[ConfigurationMetadataOwner]
+
+        :param owner_type: Owner's class
+        :type owner_type: Optional[Type]
```

Comment:
`owner_type` isn't used and shouldn't be necessary since you can call `type(owner_instance)`.

Reply:
Never mind -- this is part of the `__get__` contract.